File: 27. Remove Element/remove_element.py
def remove_by_shifting (element, array):
	count = 0
	len_array = len(array)
	
	for i in range(len_array - 1, -1, -1):
		if (array[i] == element):
			count += 1
			#shift code
			holder = array[i]
			
			for j in range (i, (len_array - 1)):
				array[j] = array[j + 1]
				
			array[len_array - 1] = holder
			
	return count

nums = [0,1,2,2,3,0,4,2]
val = 2

#nums = [3,2,2,3]
#val = 3

rem = remove_by_shifting (val, nums)
print(rem)
print("\n\n\n\n")

print(nums)
print(nums[:-rem])

File: 27. Remove Element/remove_element.php
<?php
function removeByShifting ($element, &$list){
	$count = 0;
	$lenList = count($list);
	
	for ($i = ($lenList - 1); $i >= 0; $i--){
		if ($list[$i] == $element){
			$count += 1;
			$holder = $list[$i];
			
			for ($j = $i; $j < ($lenList - 1); $j++){
				$list[$j] = $list[$j + 1];
			}
			
			$list[$lenList - 1] = $holder;
		}
	}
	
	return ($count);
}

$nums = [0,1,2,2,3,0,4,2];
$val = 2;

$rem = removeByShifting($val, $nums);

echo ($rem . "\n\n\n" . json_encode($nums) . "\n" . json_encode(array_slice($nums, 0, (count($nums) - $rem))) . "\n\n\n");
?>
